### PR TITLE
feat: go collector and process collector

### PR DESCRIPTION
**Pull request by yogyrahmawan**
## What does this PR do?
* added go collector and process collector metrics

## Why are we doing this? Any context or related work?
_You may put your JIRA card link here_

## Where should a reviewer start?
_optional -- if your changes affected so much files, it is encouraged to give helper for reviewer_

## Screenshots
_optional -- You may put the database, sequence or any diagram needed_

## Manual testing steps?
_Steps to do tests. including all possible that can hape_

## Database changes
_optional -- If there's database changes, put it here_

## Config changes
_optional -- If there's config changes, put it here_

## Deployment instructions
_optional -- Better to put it if there's some 'special case' for deployment_

### CODE REVIEW

File: metrics/collector/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Perkakas Collector
+This library helps you to send go collector and process collector backed by statsd
+
+# Component 
+* go_collector: go collector including goroutine, gc, etc
+* process_collector: process collector including virtual mem, resident mem, and file descriptor. only available in linux since it reads from `/proc`
```
[the remark this answers is not in the document]
Comment (yogyrahmawan) [reply metrics/collector/README.md:6]: Engga mas @ardityawahyu , udah gw add guard https://github.com/kitabisa/perkakas/blob/feat_metrics_collector/metrics/collector/process_collector.go#L153 . os x ga provide /proc

**Review by ardityawahyu — APPROVED** [review]

**Review by nabePi — APPROVED** [review]

File: metrics/collector/go_collector.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+		for _, cFn := range collectorFuncs {
+			cFn()
+		}
+	}
+}
+
+func (g *GoCollector) collectNumOfGoroutine(
+	gaugeFn func(name string, value float64, tags []string, rate float64) error,
+	_ func(name string, value int64, tags []string, rate float64) error,
+) func() {
```
Comment (budiariyanto) [inline metrics/collector/go_collector.go:154]: iki function e digawe type anyar wae pak ben le moco ra mumet @yogyrahmawan

Comment (yogyrahmawan) [reply metrics/collector/go_collector.go:154]: nggih mas @budiariyanto , kulo implement rumiyin nggih

Comment (yogyrahmawan) [reply metrics/collector/go_collector.go]: sampun nggih mas @budiariyanto . monggo dipersani

**Review by retnowid — APPROVED** [review]

**Review by budiariyanto — APPROVED** [review]

**Pull request merged**